hermes/src/network/rpc/rest.rs

```diff
@@ -1,35 +1,102 @@
+use {
+    crate::{
+        db::{
+            Db,
+            RequestTime,
+        },
+        proof_store::{
+            Proof,
+            ProofType,
+        },
+    },
+    pyth_sdk::{
+        PriceFeed,
+        PriceIdentifier,
+    },
+};
 // This file implements a REST service for the Price Service. This is a mostly direct copy of the
 // TypeScript implementation in the `pyth-crosschain` repo. It uses `axum` as the web framework and
 // `tokio` as the async runtime.
 use {
     anyhow::Result,
     axum::{
-        extract::{
-            Query,
-            State,
+        extract::State,
+        http::StatusCode,
+        response::{
+            IntoResponse,
+            Response,
         },
-        response::IntoResponse,
         Json,
     },
+    axum_extra::extract::Query, // Axum extra Query allows us to parse multi-value query parameters.
 };
 
+pub enum RestError {
+    InvalidPriceId,
+    ProofNotFound,
+}
+
+impl IntoResponse for RestError {
+    fn into_response(self) -> Response {
+        match self {
+            RestError::InvalidPriceId => {
+                (StatusCode::BAD_REQUEST, "Invalid Price Id").into_response()
+            }
+            RestError::ProofNotFound => (StatusCode::NOT_FOUND, "Proof not found").into_response(),
+        }
+    }
+}
+
 #[derive(Debug, serde::Serialize, serde::Deserialize)]
 pub struct LatestVaaQueryParams {
     ids: Vec<String>,
 }
 
-/// REST endpoint /latest_price_feeds?ids[]=...&ids[]=...&ids[]=...
-/// TODO: Replace Infallible with an actual error return type instead of unwrap() crashing the RPC.
+/// REST endpoint /latest_vaas?ids[]=...&ids[]=...&ids[]=...
+///
+/// TODO: This endpoint returns proof as a byte array. We should probably return a base64 or hex string.
```

Comment:
suggest making it configurable with a get parameter

Reply:
Will do in another PR that works on this API more specifically.